decoder/src/decoder.rs

```diff
@@ -248,15 +224,9 @@ impl<'t, 'b> Decoder<'t, 'b> {
                 Type::Usize => args.push(Arg::Uxx(read_leb128(&mut self.bytes)? as u128)),
                 Type::F32 => args.push(Arg::F32(f32::from_bits(self.bytes.read_u32::<LE>()?))),
                 Type::F64 => args.push(Arg::F64(f64::from_bits(self.bytes.read_u64::<LE>()?))),
-                Type::Bool => {
-                    let arc = Arc::new(Bool::FALSE);
-                    args.push(Arg::Bool(arc.clone()));
-                    self.bools_tbd.push(arc.clone());
-                    if self.bools_tbd.len() == MAX_NUM_BOOL_FLAGS {
-                        // reached end of compression block: sprinkle values into args
-                        self.read_and_unpack_bools()?;
-                    }
-                }
+                Type::Bool => args.push(Arg::Bool(Arc::new(Bool(AtomicBool::new(
+                    self.bytes.read_u8()? != 0,
```

Comment:
I would prefer to make it an error if this is neither 0 nor 1, that gives us more ways to detect a malformed stream, and leaves some unassigned values for later optimizations

Reply:
Makes a lot of sense, done! I also removed the Arc.